apps/remix-ide/src/app/tabs/settings-tab.js

```diff
@@ -199,29 +205,35 @@ module.exports = class SettingsTab extends ViewPlugin {
       self.config.set('settings/generate-contract-metadata', !isChecked)
       elementStateChanged(self._view.generateContractMetadataLabel, isChecked)
     }
+
     function onchangeOption (event) {
       const isChecked = self.config.get('settings/always-use-vm')
 
       self.config.set('settings/always-use-vm', !isChecked)
       elementStateChanged(self._view.optionVMLabel, isChecked)
     }
+
     function textWrapEvent (event) {
       const isChecked = self.config.get('settings/text-wrap')
 
       self.config.set('settings/text-wrap', !isChecked)
       elementStateChanged(self._view.textWrapLabel, isChecked)
       self.editor.resize(!isChecked)
     }
+
     function onchangePersonal (event) {
       const isChecked = self.config.get('settings/personal-mode')
 
       self.config.set('settings/personal-mode', !isChecked)
       elementStateChanged(self._view.personalLabel, isChecked)
     }
+
     function onchangeMatomoAnalytics (event) {
-      elementStateChanged(self._view.useMatomoAnalyticsLabel, event.target.checked)
-      const _paq = window._paq = window._paq || []
-      if (this.checked) {
+      const isChecked = self.config.get('settings/matomo-analytics')
+
+      self.config.set('settings/matomo-analytics', !isChecked)
```

Comment:
out of curiosity, any reason you need to save `!isChecked` instead of `isChecked`

Reply:
to be honest, the other way around would be probably better but it was done like that for other choices I decided to not brake consistency